api/src/pcapi/routes/backoffice/accounts/blueprint.py

```diff
@@ -1291,3 +1294,23 @@ def has_gdpr_extract(user: users_models.User) -> bool:
     if not user.gdprUserDataExtract:
         return False
     return any(not extract.is_expired for extract in user.gdprUserDataExtract)
+
+
+@public_accounts_blueprint.route("/<int:user_id>/invalidate-password", methods=["POST"])
+@atomic()
+@utils.permission_required(perm_models.Permissions.MANAGE_PUBLIC_ACCOUNT)
+def invalidate_public_account_password(user_id: int) -> utils.BackofficeResponse:
+    user = users_models.User.query.filter(users_models.User.id == user_id).one_or_none()
+    if not user:
+        raise NotFound()
+    if not (user.is_beneficiary or user.roles == []):
+        flash("Seul le mot de passe d'un compte bénéficiaire peut être invalidé", "warning")
```

Comment:
Comme tu as mis à jour la condition, il faut aussi ajuster le message d'erreur par cohérence : 
```suggestion
        flash("Seul le mot de passe d'un compte bénéficiaire ou grand public peut être invalidé", "warning")
```